group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -719,20 +721,26 @@ public void run() {
                                 result.records()
                             );
 
-                            timer.add(new TimerTask(timeout) {
+                            timer.add(new TimerTask(timeout.toMillis()) {
                                 @Override
                                 public void run() {
-                                    scheduleInternalOperation(name, tp, () -> complete(new TimeoutException("Writing records to the log timed out")));
+                                    if (!future.isDone()) {
+                                        scheduleInternalOperation(
+                                            "LogAppendEvent(name=" + name + ", tp=" + tp + ")",
+                                            tp,
+                                            () -> complete(new TimeoutException("Log append event " + name + "timed out for TopicPartition " + tp))
+                                        );
+                                    }
                                 }
                             });
 
-                            context.coordinator.updateLastWrittenOffset(offset);
-
                             // Add the response to the deferred queue.
                             if (!future.isDone()) {
                                 context.deferredEventQueue.add(offset, this);
+                                context.coordinator.updateLastWrittenOffset(offset);
```

Comment:
Why did you move this? I think that it should stay where it was and the timer must be added here, no?

Reply:
Ok. yes, that was a mistake.